src/dropdown/dropdown.js

```diff
@@ -264,9 +247,57 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
       if (!dropdownCtrl.dropdownMenu) {
         dropdownCtrl.dropdownMenu = element;
       }
+
+      var dropdown, appendToBody,
+          dropdownMenu = element;
+
+      scope.$watch(dropdownCtrl.isOpen, function( isOpen ) {
```

Comment:
@wesleycho This is purely DOM logic (element positioning and adding/removing classes) with nothing to do with $scope, model, etc. 

From the [official Directives guide](https://docs.angularjs.org/guide/directive) that provides example with `scope.$watch` in the link function: 
> "Directives that want to modify the DOM typically use the link option."

Why do you think it belongs in the controller instead of link function?